### PR TITLE
[SPARK-48027][SQL][FOLLOWUP] Add comments for the other code branch

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR propose to add comments for the other code branch.

### Why are the changes needed?
https://github.com/apache/spark/pull/46263 missing the comments for the other code branch.

### Does this PR introduce _any_ user-facing change?
'No'.

### How was this patch tested?
N/A

### Was this patch authored or co-authored using generative AI tooling?
'No'.

### CODE REVIEW

**Comment by beliefer** [comment]
ping @AngersZhuuuu cc @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InjectRuntimeFilter.scala [hunk 1]
```diff
@@ -152,7 +152,15 @@ object InjectRuntimeFilter extends Rule[LogicalPlan] with PredicateHelper with J
...
-            // We can also extract from the left side if the join keys are transitive.
+            // We can also extract from the left side if the join keys are transitive, and
+            // the left side always produces a superset output of join right keys.
+            // Let's look at an example
+            //     left table: 1, 2, 3
+            //     right table, 3, 4
+            //     right outer join output: (3, 3), (null, 4)
+            //     right key output: 3, 4
+            // Any join side always produce a superset output of its corresponding
+            // join keys, but for transitive join keys we need to check the join type.
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InjectRuntimeFilter.scala:163]: shall we refine and move the comments to the beginning? Then we don't need to duplicate it

Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InjectRuntimeFilter.scala]: Done

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by beliefer** [comment]
@cloud-fan Thank you!

**Pull request closed without merging**